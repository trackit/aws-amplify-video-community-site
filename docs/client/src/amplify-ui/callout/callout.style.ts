import {css} from "emotion";

export const hostStyle = css`
  display: block;
  padding-left: 0.75rem;
  margin-bottom: 1rem;

  > div {
    padding: 1rem;
    background-color: rgba(255, 255, 255, 0.75);
  }

  p {
    margin-bottom: 0;
    margin-top: 1rem;

    &:first-child {
      margin-top: 0;
    }
  }
`;

export const styleByType = {
  info: css`
    background-color: var(--color-ink-hv);
  `,
  warning: css`
    background-color: var(--color-orange-hv);
  `,
};
