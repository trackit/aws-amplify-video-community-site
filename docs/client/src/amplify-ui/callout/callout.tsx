import {Component, h, Prop, Host} from "@stencil/core";
import {hostStyle, styleByType} from "./callout.style";

@Component({tag: "amplify-callout", shadow: false})
export class AmplifyCallout {
  /*** is it a warning? */
  @Prop() readonly warning?: boolean;

  type?: "error" | "warning";

  componentWillLoad() {
    this.type = this.warning ? "warning" : undefined;
  }

  render() {
    return (
      <Host
        class={{
          [hostStyle]: true,
          [styleByType[this.type || "info"]]: true,
        }}
      >
        <div>
          <slot />
        </div>
      </Host>
    );
  }
}
