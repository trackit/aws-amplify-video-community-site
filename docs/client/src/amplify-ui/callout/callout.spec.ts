import {AmplifyCallout} from "./callout";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-callout", () => {
  describe("Component logic", () => {
    let callout: AmplifyCallout;
    beforeEach(() => (callout = new AmplifyCallout()));

    it("should init `warning` as `undefined`", () =>
      expect(callout.warning).toBeUndefined());

    it("should init `type` as `undefined`", () =>
      expect(callout.type).toBeUndefined());
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifyCallout],
            html: `<amplify-callout />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
