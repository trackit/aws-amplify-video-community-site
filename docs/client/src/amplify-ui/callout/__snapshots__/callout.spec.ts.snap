// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-callout Render logic should render 1`] = `
<amplify-callout class="css-wt2bt2 css-xtlbgq">
  <!---->
  <div></div>
</amplify-callout>
`;
